### PR TITLE
usbtop: init at 1.0

### PR DESCRIPTION
**Pull request by etu**
###### Motivation for this change
usbtop is a top-like utility for displaying traffic on USB buses.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/usbtop/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    rev = "release-${version}";
+    sha256 = "0qbad0aq6j4jrh90l6a0akk71wdzhyzmy6q8wl138axyj2bp9kss";
+  };
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ libpcap boost ];
+
+  meta = with stdenv.lib; {
+    homepage = "https://github.com/aguinet/usbtop";
+    description = "top-like utility that shows an estimated instantaneous bandwidth on USB buses and devices";
```
Comment (alexarice) [inline pkgs/os-specific/linux/usbtop/default.nix]: Description should be capitalised

Comment (ivan) [reply pkgs/os-specific/linux/usbtop/default.nix]: Some packages solve this problem by writing "A top-like" or "A 'top'"

**Comment by ivan** [comment]
This builds and runs for me on NixOS x86_64, but then I continued testing by repeatedly starting `usbtop` during a `btrfs scrub start ...` on a problem-free Samsung MUF 256GB, and got:

```
Aug 30 15:34:34 plato kernel: usb 4-2: usbfs: USBDEVFS_CONTROL failed cmd usbtop rqt 128 rq 6 len 273 ret -110
Aug 30 15:34:45 plato kernel: usb 4-2: usbfs: USBDEVFS_CONTROL failed cmd usbtop rqt 128 rq 6 len 273 ret -110
Aug 30 15:34:51 plato kernel: usb 4-2: usbfs: USBDEVFS_CONTROL failed cmd usbtop rqt 128 rq 6 len 273 ret -110
Aug 30 15:35:02 plato kernel: xhci_hcd 0000:26:00.0: Timeout while waiting for setup device command
Aug 30 15:35:02 plato kernel: usb 4-2: Device not responding to setup address.
Aug 30 15:35:03 plato kernel: usb 4-2: device not accepting address 2, error -71
Aug 30 15:35:03 plato kernel: usb 4-2: Device not responding to setup address.
Aug 30 15:35:03 plato kernel: usb 4-2: Device not responding to setup address.
Aug 30 15:35:03 plato kernel: usb 4-2: device not accepting address 2, error -71
Aug 30 15:35:03 plato kernel: usb 4-2: Device not responding to setup address.
Aug 30 15:35:04 plato kernel: usb 4-2: Device not responding to setup address.
Aug 30 15:35:04 plato kernel: usb 4-2: device not accepting address 2, error -71
Aug 30 15:35:04 plato kernel: usb 4-2: Device not responding to setup address.
Aug 30 15:35:04 plato kernel: usb 4-2: Device not responding to setup address.
Aug 30 15:35:04 plato kernel: usb 4-2: device not accepting address 2, error -71
Aug 30 15:35:04 plato kernel: usb 4-2: USB disconnect, device number 2
Aug 30 15:35:04 plato kernel: print_req_error: I/O error, dev sdc, sector 18620416
Aug 30 15:35:04 plato kernel: print_req_error: I/O error, dev sdc, sector 18622464
Aug 30 15:35:04 plato kernel: print_req_error: I/O error, dev sdc, sector 18620416
Aug 30 15:35:04 plato kernel: BTRFS error (device dm-2): bdev /dev/mapper/Samsung-256-1 errs: wr 0, rd 1, flush 0, corrupt 0, gen 0
Aug 30 15:35:04 plato kernel: BTRFS error (device dm-2): bdev /dev/mapper/Samsung-256-1 errs: wr 0, rd 2, flush 0, corrupt 0, gen 0
Aug 30 15:35:04 plato kernel: BTRFS error (device dm-2): bdev /dev/mapper/Samsung-256-1 errs: wr 0, rd 3, flush 0, corrupt 0, gen 0
Aug 30 15:35:04 plato kernel: BTRFS error (device dm-2): unable to fixup (regular) error at logical 8448376832 on dev /dev/mapper/Samsung-256-1
Aug 30 15:35:04 plato kernel: BTRFS error (device dm-2): bdev /dev/mapper/Samsung-256-1 errs: wr 0, rd 4, flush 0, corrupt 0, gen 0
Aug 30 15:35:04 plato kernel: BTRFS error (device dm-2): bdev /dev/mapper/Samsung-256-1 errs: wr 0, rd 5, flush 0, corrupt 0, gen 0
Aug 30 15:35:04 plato kernel: BTRFS error (device dm-2): bdev /dev/mapper/Samsung-256-1 errs: wr 0, rd 6, flush 0, corrupt 0, gen 0
Aug 30 15:35:04 plato kernel: BTRFS error (device dm-2): unable to fixup (regular) error at logical 8448380928 on dev /dev/mapper/Samsung-256-1
Aug 30 15:35:04 plato kernel: BTRFS error (device dm-2): bdev /dev/mapper/Samsung-256-1 errs: wr 0, rd 7, flush 0, corrupt 0, gen 0
Aug 30 15:35:04 plato kernel: BTRFS error (device dm-2): bdev /dev/mapper/Samsung-256-1 errs: wr 0, rd 8, flush 0, corrupt 0, gen 0
Aug 30 15:35:04 plato kernel: BTRFS error (device dm-2): bdev /dev/mapper/Samsung-256-1 errs: wr 0, rd 9, flush 0, corrupt 0, gen 0
Aug 30 15:35:04 plato kernel: BTRFS error (device dm-2): unable to fixup (regular) error at logical 8448385024 on dev /dev/mapper/Samsung-256-1
Aug 30 15:35:04 plato kernel: BTRFS error (device dm-2): bdev /dev/mapper/Samsung-256-1 errs: wr 0, rd 10, flush 0, corrupt 0, gen 0
Aug 30 15:35:04 plato kernel: BTRFS error (device dm-2): unable to fixup (regular) error at logical 8448389120 on dev /dev/mapper/Samsung-256-1
Aug 30 15:35:04 plato kernel: BTRFS error (device dm-2): unable to fixup (regular) error at logical 8448393216 on dev /dev/mapper/Samsung-256-1
Aug 30 15:35:04 plato kernel: BTRFS error (device dm-2): unable to fixup (regular) error at logical 8448397312 on dev /dev/mapper/Samsung-256-1
Aug 30 15:35:04 plato kernel: BTRFS error (device dm-2): unable to fixup (regular) error at logical 8448401408 on dev /dev/mapper/Samsung-256-1
Aug 30 15:35:04 plato kernel: BTRFS error (device dm-2): unable to fixup (regular) error at logical 8448405504 on dev /dev/mapper/Samsung-256-1
Aug 30 15:35:04 plato kernel: BTRFS error (device dm-2): unable to fixup (regular) error at logical 8448409600 on dev /dev/mapper/Samsung-256-1
Aug 30 15:35:04 plato kernel: BTRFS error (device dm-2): unable to fixup (regular) error at logical 8448413696 on dev /dev/mapper/Samsung-256-1
Aug 30 15:35:04 plato kernel: scsi 6:0:0:0: rejecting I/O to dead device
Aug 30 15:35:04 plato kernel: print_req_error: I/O error, dev sdc, sector 18620672
Aug 30 15:35:05 plato kernel: scsi 6:0:0:0: rejecting I/O to dead device
Aug 30 15:35:05 plato kernel: print_req_error: I/O error, dev sdc, sector 18620928
Aug 30 15:35:05 plato kernel: scsi 6:0:0:0: rejecting I/O to dead device
Aug 30 15:35:05 plato kernel: print_req_error: I/O error, dev sdc, sector 18624512
Aug 30 15:35:05 plato kernel: scsi 6:0:0:0: rejecting I/O to dead device
Aug 30 15:35:05 plato kernel: print_req_error: I/O error, dev sdc, sector 18621184
Aug 30 15:35:05 plato kernel: scsi 6:0:0:0: rejecting I/O to dead device
Aug 30 15:35:05 plato kernel: print_req_error: I/O error, dev sdc, sector 18621440
Aug 30 15:35:05 plato kernel: scsi 6:0:0:0: rejecting I/O to dead device
Aug 30 15:35:05 plato kernel: print_req_error: I/O error, dev sdc, sector 18621696
Aug 30 15:35:05 plato kernel: scsi 6:0:0:0: rejecting I/O to dead device
Aug 30 15:35:05 plato kernel: print_req_error: I/O error, dev sdc, sector 18626560
Aug 30 15:35:05 plato usbguard-daemon[829]: [1567179305.077] (W) Ignoring unknown UEvent action: sysfs_devpath=/devices/pci0000:00/0000:00:1c.7/0000:26:00.0/usb4/4-2/4-2:1.0 action=unbind
Aug 30 15:35:05 plato usbguard-daemon[829]: [1567179305.094] (W) Ignoring unknown UEvent action: sysfs_devpath=/devices/pci0000:00/0000:00:1c.7/0000:26:00.0/usb4/4-2 action=unbind
Aug 30 15:35:05 plato usbguard-daemon[829]: [1567179305.096] (A) uid=0 pid=829 device.rule='allow id 090c:1000 serial "0347018060001560" name "Flash Drive FIT" hash "4J1o+D9D9x5kCDQFzzQtlD9SHKZZ9/aASnNPYOLcqnQ=" parent-hash ">
Aug 30 15:35:05 plato kernel: BTRFS: error (device dm-2) in btrfs_commit_transaction:2260: errno=-5 IO failure (Error while writing out transaction)
Aug 30 15:35:05 plato kernel: BTRFS info (device dm-2): forced readonly
Aug 30 15:35:05 plato kernel: BTRFS warning (device dm-2): Skipping commit of aborted transaction.
Aug 30 15:35:05 plato kernel: BTRFS: error (device dm-2) in cleanup_transaction:1860: errno=-5 IO failure
Aug 30 15:35:05 plato kernel: BTRFS info (device dm-2): delayed_refs has NO entry
Aug 30 15:35:05 plato kernel: usb 4-2: Device not responding to setup address.
Aug 30 15:35:05 plato kernel: usb 4-2: Device not responding to setup address.
Aug 30 15:35:05 plato kernel: usb 4-2: device not accepting address 3, error -71
Aug 30 15:35:05 plato kernel: usb 4-2: Device not responding to setup address.
Aug 30 15:35:06 plato kernel: usb 4-2: Device not responding to setup address.
Aug 30 15:35:06 plato kernel: usb 4-2: device not accepting address 4, error -71
Aug 30 15:35:06 plato kernel: usb usb4-port2: attempt power cycle
Aug 30 15:35:11 plato kernel: xhci_hcd 0000:26:00.0: Error while assigning device slot ID
Aug 30 15:35:11 plato kernel: xhci_hcd 0000:26:00.0: Max number of devices this xHCI host supports is 32.
Aug 30 15:35:11 plato kernel: usb usb4-port2: couldn't allocate usb_device
```

usbtop-btrfs-scrub-50

File: pkgs/os-specific/linux/usbtop/default.nix (hunk 1 above)
Comment (etu) [reply pkgs/os-specific/linux/usbtop/default.nix]: Fixed

**Comment by etu** [comment]
@ivan Sounds more... like a bug with this program, btrfs or a resource limitation of some kind. Maybe it should be reported somewhere?

**Comment by aanderse** [comment]
@etu you've run this a bit with no problems? Everything looks good. Ready for a merge?

**Comment by etu** [comment]
@aanderse Yeah, works fine for me here. So I don't think it's a packaging issue.

**Review by aanderse — APPROVED** [review]
LGTM :+1:

**Pull request merged**